pkg/pillar/types/zedagenttypes.go

```diff
@@ -207,3 +207,40 @@ type DatastoreConfig struct {
 func (config DatastoreConfig) Key() string {
 	return config.UUID.String()
 }
+
+// NodeAgentStatus :
+type NodeAgentStatus struct {
+	Name              string
+	CurPart           string
+	TestInprogress    bool
+	UpdateInprogress  bool
+	RemainingTestTime time.Duration
+	NeedsReboot       bool
+	TestComplete      bool
+	RebootReason      string
+	ErrorStr          string
+}
+
+// Key :
+func (status NodeAgentStatus) Key() string {
+	return status.Name
+}
+
+type ConfigGetStatus uint8
+
+const (
+	CONFIG_GET_SUCCESS ConfigGetStatus = iota + 1
+	CONFIG_GET_FAIL
+	CONFIG_GET_FAIL_400
```

Comment:
Now or later we should add a CONFIG_GET_READ_SAVED to indicate that the config exists, but is from a checkpoint. That is useful to start the GC in other agents.

Reply:
Have added, the enum and handling inside zedagent code, for  ConfigGetReadSaved status.